File: 2447-number-of-subarrays-with-gcd-equal-to-k/2447-number-of-subarrays-with-gcd-equal-to-k.py
class Solution:
    def subarrayGCD(self, nums: List[int], k: int) -> int:
        length = len(nums)
        ans = 0
        for i in range(length):
            temp = nums[i]
            for j in range(i, length):
                temp = math.gcd(temp, nums[j])
                if temp == k:
                    ans += 1
                elif temp < k:
                    break
        return ans

File: 2447-number-of-subarrays-with-gcd-equal-to-k/2447-number-of-subarrays-with-gcd-equal-to-k.cpp
class Solution {
public:
    int subarrayGCD(vector<int>& nums, int k) {
        int ans = 0;
        int temp = 0;
        int i, j;
        int n = nums.size();
        
        for(i = 0; i < n; i++){
            temp = nums[i];
            for(j = i; j < n; j++){
                temp = __gcd(temp, nums[j]);

                if(temp == k){
                    ans++;
                }
                
            }
        }

        return ans;
    }
};